src/main/scala/chisel3/tester/Testers2OptionsManager.scala

```diff
@@ -0,0 +1,17 @@
+// See LICENSE for license details.
```

Comment:
Is this used enough to justify creating an OptionsManager class? Is it sufficiently general? For example, what happens if we add Verilator or VCS support?

My current preference is to provide options as a "escape hatch" only, and zero-configuration (or maybe parsing of parameters passed to ScalaTest in infrastructure) should handle most cases. I know that Treadle needs a registry of BlackBoxFactories, but I think creating a `new ExecutionOptionsManager with HasTreadleSuite with ...` at the use site is fine for now.

Also from an API standpoint I don't want to give the impression that spamming OptionsManager is encouraged.

Reply:
It is definitely worth creating the class, just to be DRY and also to manage the rat's nest of traits involved. I don't think this should be a sticking point as this will all be re-done more rationally
with the Options As Annotations project.

Reply:
- I made this class after spending way to much time trying to the `with HasXXXOptions` right in order to get the options manager passed down to treadle so I could see the options. 
- I don't want to go through gymnastics of the raw API here. It's not worth the time
- It's not DRY to write all those `with`s
- OptionsAsAnnotations will explicitly remove the current options managers so this will definitely go away